src/main/java/com/achdev/onlinebookstoreapp/mapper/role/RoleSetMapper.java

```diff
@@ -0,0 +1,26 @@
+package com.achdev.onlinebookstoreapp.mapper.role;
+
+import com.achdev.onlinebookstoreapp.model.Role;
+import com.achdev.onlinebookstoreapp.service.RoleService;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class RoleSetMapper {
```

Comment:
why do you need this mapper?

Reply:
I thought it would be needed.

Reply:
If I remove it, then user's role data don't store in the user_roles table.

Reply:
And that's why I added "private list <Long> roles"  in the  "UserRegistrationRequestDto" class.